tests/SelfTest/UsageTests/MatchersRanges.tests.cpp

```diff
@@ -548,3 +554,210 @@ TEST_CASE( "The quantifier range matchers support types with different types ret
+            return std::abs( lhs - rhs ) <= 1;
+        };
+
+        SECTION( "Two equal non-empty containers (close enough)" ) {
+            const std::vector<int> vector_a{ { 1, 10, 20 } };
+            const std::vector<int> vector_a_plus_1{ { 11, 21, 2 } };
+            CHECK_THAT( vector_a,
+                        UnorderedElementsAre( vector_a_plus_1, close_enough ) );
+        }
+        SECTION( "Two non-equal non-empty containers (close enough)" ) {
+            const std::vector<int> vector_a{ { 1, 10, 21 } };
+            const std::vector<int> vector_b{ { 11, 21, 3 } };
+            CHECK_THAT( vector_a,
+                        !UnorderedElementsAre( vector_b, close_enough ) );
+        }
+    }
+
+    // As above with ElementsAre, short cicuiting and other optimisations
+    // are left to the STL implementation
+}
+
+TEST_CASE( "Type conversions of ElementsAre and similar",
+           "[matchers][templated][quantifiers]" ) {
+    using Catch::Matchers::ElementsAre;
+    using Catch::Matchers::UnorderedElementsAre;
+
+    // In these test, we can always test ElementsAre and
+    // UnorderedElementsAre in the same way, since we're mostly
+    // testing the template type deductions (and ElementsAre
+    // implies UnorderedElementsAre)
+
+    SECTION( "Container conversions" ) {
+        SECTION( "Two equal containers of different container types" ) {
+            const std::array<int, 3> array_int_a{ { 1, 2, 3 } };
+            const std::vector<int> vector_int_a{ 1, 2, 3 };
+            CHECK_THAT( array_int_a, ElementsAre( vector_int_a ) );
+            CHECK_THAT( array_int_a, UnorderedElementsAre( vector_int_a ) );
+        }
+        SECTION( "Two equal containers of different container types "
+                    "(differ in array N)" ) {
+            const std::array<int, 3> array_int_3{ { 1, 2, 3 } };
+            const std::array<int, 4> array_int_4{ { 1, 2, 3, 4 } };
+            CHECK_THAT( array_int_3, !ElementsAre( array_int_4 ) );
+            CHECK_THAT( array_int_3, !UnorderedElementsAre( array_int_4 ) );
+        }
+        SECTION( "Two equal containers of different container types and value "
+                    "types" ) {
+            const std::array<int, 3> array_int_a{ { 1, 2, 3 } };
+            const std::vector<int> vector_char_a{ 1, 2, 3 };
+            CHECK_THAT( array_int_a, ElementsAre( vector_char_a ) );
+            CHECK_THAT( array_int_a, UnorderedElementsAre( vector_char_a ) );
+        }
+        SECTION( "Two equal containers, one random access, one not" ) {
+            const std::array<int, 3> array_int_a{ { 1, 2, 3 } };
+            const std::list<int> list_char_a{ 1, 2, 3 };
+
+#ifdef CONCEPT_TESTING
+            // Verify these types really are different in random access nature
+            static_assert( std::random_access_iterator<decltype(array_int_a)::iterator> );
+            static_assert( !std::random_access_iterator<decltype(list_char_a)::iterator> );
+#endif
```

Comment:
Is there a reason not to have this in C++14 through `std::is_base_of` and `std::iterator_traits`?

Reply:
Because I was in C++20 mode and didn't think of it? 😅

Reply:
Yeah, we still do things the hard way here, C++14 only 😃